src/style/dimension.rs

```diff
@@ -0,0 +1,306 @@
+//! Style types for representing lengths / sizes
+
+use crate::geometry::{Rect, Size};
+use crate::style_helpers::{FromPercent, FromPoints, TaffyAuto, TaffyMaxContent, TaffyMinContent, TaffyZero};
+use crate::sys::abs;
+
+/// A unit of linear measurement
+///
+/// This is commonly combined with [`Rect`], [`Point`](crate::geometry::Point) and [`Size<T>`].
+#[derive(Copy, Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum LengthPercentage {
+    /// Points are abstract absolute units. Users of Taffy may define what they correspond
+    /// to in their application (pixels, logical pixels, mm, etc) as they see fit.
+    Points(f32),
+    /// The dimension is stored in percentage relative to the parent item.
+    Percent(f32),
+}
+impl TaffyZero for LengthPercentage {
+    const ZERO: Self = Self::Points(0.0);
+}
+impl FromPoints for LengthPercentage {
+    fn from_points<Input: Into<f32> + Copy>(points: Input) -> Self {
+        Self::Points(points.into())
+    }
+}
+impl FromPercent for LengthPercentage {
+    fn from_percent<Input: Into<f32> + Copy>(percent: Input) -> Self {
+        Self::Points(percent.into())
+    }
+}
+
+/// A unit of linear measurement
+///
+/// This is commonly combined with [`Rect`], [`Point`](crate::geometry::Point) and [`Size<T>`].
+#[derive(Copy, Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum LengthPercentageAuto {
+    /// Points are abstract absolute units. Users of Taffy may define what they correspond
+    /// to in their application (pixels, logical pixels, mm, etc) as they see fit.
+    Points(f32),
+    /// The dimension is stored in percentage relative to the parent item.
+    Percent(f32),
+    /// The dimension should be automatically computed
+    Auto,
+}
+impl TaffyZero for LengthPercentageAuto {
+    const ZERO: Self = Self::Points(0.0);
+}
+impl TaffyAuto for LengthPercentageAuto {
+    const AUTO: Self = Self::Auto;
+}
+impl FromPoints for LengthPercentageAuto {
+    fn from_points<Input: Into<f32> + Copy>(points: Input) -> Self {
+        Self::Points(points.into())
+    }
+}
+impl FromPercent for LengthPercentageAuto {
+    fn from_percent<Input: Into<f32> + Copy>(percent: Input) -> Self {
+        Self::Points(percent.into())
+    }
+}
+
+impl From<LengthPercentage> for LengthPercentageAuto {
+    fn from(input: LengthPercentage) -> Self {
+        match input {
+            LengthPercentage::Points(value) => Self::Points(value),
+            LengthPercentage::Percent(value) => Self::Percent(value),
+        }
+    }
+}
+
+/// A unit of linear measurement
+///
+/// This is commonly combined with [`Rect`], [`Point`](crate::geometry::Point) and [`Size<T>`].
+#[derive(Copy, Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum Dimension {
```

Comment:
This doesn't seem distinct from `LengthPercentageAuto`? This name is way nicer though!

Reply:
It's not currently, but properties using `Dimension` support additional variants like `MinContent`, `MaxContent`, and `FitContent` that we don't support yet (but will probably want to at some point). As of the next release I wouldn't expect people to be using these types directly in any case. There was discussion of this on https://github.com/DioxusLabs/taffy/pull/269 where these types were introduced.